--- origsrc/gwget-1.0.1/data/Makefile.am	2009-02-06 13:19:51.000000000 -0600
+++ src/gwget-1.0.1/data/Makefile.am	2009-02-18 14:09:42.606000000 -0600
@@ -30,9 +30,13 @@ desktop_DATA     =$(desktop_in_files:.de
 
 @INTLTOOL_DESKTOP_RULE@
 
+if GCONF_SCHEMAS_INSTALL
+
 install-data-local:
 	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule gwget.schemas
 
+endif
+
 EXTRA_DIST = $(gwget_data_DATA) \
 	     $(service_in_files) \
 	     $(desktop_in_files) \
