--- origsrc/gwget-0.99/configure.in	2007-05-28 12:32:40.000000000 -0500
+++ src/gwget-0.99/configure.in	2008-06-11 04:03:38.875000000 -0500
@@ -101,9 +101,11 @@
                        [Whether to use epiphany version 1.6 or 1.8 or 1.10 or 2.14 (default 1.6)]),
         [EPHY_VER="$withval"])
 
-dnl try to autodetect Epiphany 1.2, 1.4, 1.6, 1.8, 1.10, 2.14, 2.15, 2.16, 2.17 or 2.18
+dnl try to autodetect Epiphany 1.2, 1.4, 1.6, 1.8, 1.10, 2.14, 2.15, 2.16, 2.17, 2.18, or 2.22
 EPHY_AUTODETECT="(autodetect)"
-if pkg-config --exists epiphany-2.18; then
+if pkg-config --exists epiphany-2.22; then
+	EPHY_VER=2.22
+elif pkg-config --exists epiphany-2.18; then
 	EPHY_VER=2.18
 elif pkg-config --exists epiphany-2.17; then
         EPHY_VER=2.17
@@ -170,6 +172,10 @@
 	  EPIPHANY_PKGCONFIG=epiphany-2.18
           EPIPHANY_REQUIRED=2.18
 	  EPIPHANY_SHORTVER=218
+elif test "x$EPHY_VER" = "x2.22"; then
+          EPIPHANY_PKGCONFIG=epiphany-2.22
+          EPIPHANY_REQUIRED=2.22
+          EPIPHANY_SHORTVER=222
 else
      	AC_MSG_RESULT([no epiphany installed])
 fi
--- origsrc/gwget-0.99/data/gwget.glade	2007-05-06 14:02:08.000000000 -0500
+++ src/gwget-0.99/data/gwget.glade	2008-06-11 04:07:38.812500000 -0500
@@ -103,6 +103,7 @@
                                 <property name="visible">True</property>
                                 <property name="label" translatable="yes">Remove completed</property>
                                 <property name="use_underline">True</property>
+                                <signal name="activate" handler="on_remove_completed_activate"/>
                               </widget>
                             </child>
                             <child>
@@ -110,6 +111,7 @@
                                 <property name="visible">True</property>
                                 <property name="label" translatable="yes">Remove inactive</property>
                                 <property name="use_underline">True</property>
+                                <signal name="activate" handler="on_remove_notrunning_activate"/>
                               </widget>
                             </child>
                             <child>
@@ -117,6 +119,7 @@
                                 <property name="visible">True</property>
                                 <property name="label" translatable="yes">Remove all</property>
                                 <property name="use_underline">True</property>
+                                <signal name="activate" handler="on_remove_all_activate"/>
                               </widget>
                             </child>
                           </widget>
--- origsrc/gwget-0.99/epiphany-extension/Makefile.am	2007-04-21 14:01:17.000000000 -0500
+++ src/gwget-0.99/epiphany-extension/Makefile.am	2008-06-11 04:22:40.468750000 -0500
@@ -7,6 +7,7 @@
 
 INCLUDES =				\
  	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
+    $(GNOME_CFLAGS)		\
 	-DSHARE_DIR=\"$(pkgdatadir)\"                   \
         -DEPHY_EXTENSIONS_LOCALEDIR=\"$(datadir)/locale\"  \
 	-I$(top_srcdir)/src		\
@@ -20,7 +21,12 @@
  	$(top_builddir)/src/gwget-application.h \
  	$(top_builddir)/src/gwget-application.c  
 
-libgwgetextension_la_LDFLAGS = -module -avoid-version 
+libgwgetextension_la_LDFLAGS = -module -avoid-version -no-undefined -export-symbols $(srcdir)/gwget.def
+
+libgwgetextension_la_LIBADD =		\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
+	$(GNOME_LIBS)		\
+	$(DBUS_LIBS)
 
 extensionxmldir = $(extensiondir)
 extensionxml_in_files = gwget.xml.in.in
--- origsrc/gwget-0.99/epiphany-extension/gwget.def	1969-12-31 18:00:00.000000000 -0600
+++ src/gwget-0.99/epiphany-extension/gwget.def	2008-06-11 04:22:06.062500000 -0500
@@ -0,0 +1,4 @@
+IMPORTS
+gwget_data_new = gwget.exe.gwget_data_new
+gwget_data_add_download = gwget.exe.gwget_data_add_download
+gwget_data_start_download = gwget.exe.gwget_data_start_download
--- origsrc/gwget-0.99/gwget.desktop.in	2007-04-21 14:01:19.000000000 -0500
+++ src/gwget-0.99/gwget.desktop.in	2008-06-11 04:04:32.687500000 -0500
@@ -10,7 +10,7 @@
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=Application;Network;
+Categories=Network;FileTransfer;GNOME;GTK
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gwget
 X-GNOME-Bugzilla-Component=General
